src/core/mac/sub_mac.cpp

```diff
@@ -273,7 +273,7 @@ void SubMac::HandleReceiveDone(RxFrame *aFrame, Error aError)
     if (aFrame != nullptr && aError == kErrorNone)
     {
         // Assuming the risk of the parent missing the Enh-ACK in favor of smaller CSL receive window
-        if ((mCslPeriod > 0) && aFrame->mInfo.mRxInfo.mAckedWithSecEnhAck)
+        if ((mCslPeriod > 0) && aFrame->IsVersion2015() && aFrame->GetAckRequest())
```

Comment:
Hi, I don't think this should be changed here:

- `aFrame->IsVersion2015() && aFrame->GetAckRequest())`: the frame version is 2015 and requested ACK.
- `aFrame->mInfo.mRxInfo.mAckedWithSecEnhAck`: the frame was **actually acknowledged** with a secured ACK frame.

Instead, the platform implementation should set the proper value for the `mAckedWithSecEnhAck` flag.

Reply:
In the 15.4 2015 spec, it doesn't say that the Enh-ACK with IEs must be always secured. It has to only for secured frames.

> If the Enh-Ack frame contains IEs and/or a Frame Payload and it is in response
> to a secured frame, then the Enh-Ack frame shall be secured.

Reply:
I agree with @George-Stefan. I think the Enh-Ack isn't always secured. But we should update CSL time for these frames. Thoughts? @CiaranWoodward

Reply:
> `aError == kErrorNone` implies that the ACK was sent which must be Enh-ACK from the additional checks

Yes, good point.

Reply:
I see that `GenerateEnhAck()` secures Enh-ACK based on Rx frame attributes only.
Should check whether there are IEs too: `aIeLength > 0`

https://github.com/openthread/openthread/blob/main/src/core/mac/mac_frame.cpp#L1222

Reply:
Hi @George-Stefan where exactly are you proposing this check be made?  In `GenerateEnhAck`, I don't see a call to secure the packet, unless I'm missing it.

Reply:
The Enh-ACK is set with FCF.SEC so it's encrypted later durin Tx in radio.c
```
if (aFrame.GetSecurityEnabled())
{
    fcf |= kFcfSecurityEnabled;
}
```

Based on @CiaranWoodward comment, I see a contradiction